From 42663d4b49617c17c75c347b6927630cfd5a715e Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 10 Dec 2010 02:34:49 +0000
Subject: [PATCH 2/4] alpha: Do not use -Werror for arch/alpha

The alpha headers for I/O currently have illegal combinations of
extern inline and static inline functions calling each other.  Make
the warnings non-fatal for now.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 arch/alpha/kernel/Makefile   |    2 +-
 arch/alpha/lib/Makefile      |    1 -
 arch/alpha/mm/Makefile       |    2 --
 arch/alpha/oprofile/Makefile |    2 +-
 4 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/arch/alpha/kernel/Makefile b/arch/alpha/kernel/Makefile
index 1ee9b5b..1c77b1c 100644
--- a/arch/alpha/kernel/Makefile
+++ b/arch/alpha/kernel/Makefile
@@ -4,7 +4,7 @@
 
 extra-y		:= head.o vmlinux.lds
 EXTRA_AFLAGS	:= $(KBUILD_CFLAGS)
-EXTRA_CFLAGS	:= -Werror -Wno-sign-compare
+EXTRA_CFLAGS	:= -Wno-sign-compare
 
 obj-y    := entry.o traps.o process.o init_task.o osf_sys.o irq.o \
 	    irq_alpha.o signal.o setup.o ptrace.o time.o \
diff --git a/arch/alpha/lib/Makefile b/arch/alpha/lib/Makefile
index 9b72c59..a2723cb 100644
--- a/arch/alpha/lib/Makefile
+++ b/arch/alpha/lib/Makefile
@@ -3,7 +3,6 @@
 #
 
 EXTRA_AFLAGS := $(KBUILD_CFLAGS)
-EXTRA_CFLAGS := -Werror
 
 # Many of these routines have implementations tuned for ev6.
 # Choose them iff we're targeting ev6 specifically.
diff --git a/arch/alpha/mm/Makefile b/arch/alpha/mm/Makefile
index 09399c5..ec693be 100644
--- a/arch/alpha/mm/Makefile
+++ b/arch/alpha/mm/Makefile
@@ -2,8 +2,6 @@
 # Makefile for the linux alpha-specific parts of the memory manager.
 #
 
-EXTRA_CFLAGS := -Werror
-
 obj-y	:= init.o fault.o extable.o
 
 obj-$(CONFIG_DISCONTIGMEM) += numa.o
diff --git a/arch/alpha/oprofile/Makefile b/arch/alpha/oprofile/Makefile
index 4aa5624..74cb06e 100644
--- a/arch/alpha/oprofile/Makefile
+++ b/arch/alpha/oprofile/Makefile
@@ -1,4 +1,4 @@
-EXTRA_CFLAGS := -Werror -Wno-sign-compare
+EXTRA_CFLAGS := -Wno-sign-compare
 
 obj-$(CONFIG_OPROFILE) += oprofile.o
 
-- 
1.7.2.3

